### PR TITLE
Expose wether exception cause is supported

### PR DESCRIPTION
**Pull request by JonRowe**
No changelog because this is an internal feature

### CODE REVIEW

**Comment by JonRowe** [comment]
@samphippen care to give this a once over?

File: lib/rspec/support/ruby_features.rb [hunk 1]
```diff
@@ -55,6 +55,18 @@ def caller_locations_supported?
         respond_to?(:caller_locations, true)
       end
 
+      begin
```
Comment (samphippen) [inline lib/rspec/support/ruby_features.rb]: What do you think about doing a `respond_to?` check here? I think that'd be a lot cleaner.

Comment (JonRowe) [inline lib/rspec/support/ruby_features.rb]: I preferred not instantiating an exception...

Comment (samphippen) [inline lib/rspec/support/ruby_features.rb]: oh, right, `Exception.instance_methods.include?(:cause)`?

Comment (myronmarston) [inline lib/rspec/support/ruby_features.rb]: `Exception.method_defined?(:cause)`

No need for a linear `include?` operation.

Comment (samphippen) [inline lib/rspec/support/ruby_features.rb]: :+1:

Comment (JonRowe) [inline lib/rspec/support/ruby_features.rb]: Done.

**Comment by JonRowe** [comment]
Hmm randomly this is failing on windows but it's for an unrelated encoding spec..

**Comment by myronmarston** [comment]
LGTM.  Merged.

**Pull request merged**